ui/raidboss/data/02-arr/trial/ultima-ex.ts

```diff
@@ -0,0 +1,126 @@
+      outputStrings: {
+        text: {
+          en: 'Off-tank cleave',
+        },
+      },
+    },
+    {
+      id: 'Ultima EX Vulcan Burst',
+      regex: /Vulcan Burst/,
+      beforeSeconds: 4,
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Melee knockback',
+        },
+      },
+    },
+  ],
+  triggers: [
+    {
+      id: 'Ultima EX Tank Purge',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '5EA', source: 'The Ultima Weapon', capture: false }),
+      response: Responses.bigAoe(),
+    },
+    {
+      // At 5 stacks of Viscous Aetheroplasm, the target begins taking massive damage.
+      id: 'Ultima EX Viscous Aetheroplasm',
+      type: 'GainsEffect',
+      netRegex: NetRegexes.gainsEffect({ effectId: '171', count: '04', capture: false }),
+      condition: (data) => data.role === 'tank',
+      alertText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Tank Swap Now--4 Stacks',
+        },
+      },
+    },
+    {
+      id: 'Ultima EX Homing Aetheroplasm Collect',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '672' }),
+      run: (data, matches) => {
+        data.plasmTargets = data.plasmTargets ??= [];
+        data.plasmTargets.push(matches.target);
+      },
+    },
+    {
+      // This ability is ARR 2.0's version of headmarkers. No associated 27 lines are present.
+      // These lines are sent by entities with no name and no 03/04 lines.
+      id: 'Ultima EX Homing Aetheroplasm Call',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '672', capture: false }),
+      delaySeconds: 0.5,
+      suppressSeconds: 5,
+      infoText: (data, _matches, output) => {
+        if (data.plasmTargets?.includes(data.me))
+          return output.target!();
+        return output.avoid!();
+      },
+      outputStrings: {
+        target: {
+          en: 'Homing Aetheroplasm on YOU',
+        },
+        avoid: {
+          en: 'Avoid Homing Aetheroplasm',
+        },
+      },
+    },
+    {
+      id: 'Ultima EX Homing Aetheroplasm Cleanup',
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '672', capture: false }),
+      suppressSeconds: 5,
+      run: (data) => delete data.plasmTargets,
+    },
+    {
+      // We use a StartsUsing line here because we can't use timeline triggers for this,
+      // and we want to warn players as early as possible.
+      id: 'Ultima EX Aetheric Boom',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '5E7', source: 'The Ultima Weapon', capture: false }),
+      alarmText: (data, _matches, output) => output[data.boomCounter]!(),
+      outputStrings: {
+        0: {
+          en: 'Boom Orbs: 1x at cardinals',
+        },
+        1: {
+          en: 'Boom Orbs: 1x N/S, 2x E/W',
+        },
+        2: {
+          en: 'Boom Orbs: 2x at intercardinals',
+        },
```

Comment:
Minor brevity suggestion, which you can take or leave.

```suggestion
      outputStrings: {
        0: {
          en: 'Orbs: Cardinals',
        },
        1: {
          en: 'Orbs: Cardinals (N/S first)',
        },
        2: {
          en: 'Orbs: Intercardinals',
        },
```

These "strings" 0, 1, 2 are shown to the user in the config ui.  Would it make sense to start at 1, or use `boom${data.boomCounter}` so the strings are `boom1`, `boom2`, etc?

Reply:
You're right, I think I'll combine those just to make it easier for everyone.